docker/archive/src.go

```diff
@@ -34,3 +35,83 @@ func newImageSource(ctx context.Context, sys *types.SystemContext, ref archiveRe
 func (s *archiveImageSource) Reference() types.ImageReference {
 	return s.ref
 }
+
+// MultiImageSourceItem is a reference to _one_ image in a multi-image archive.
+// Note that MultiImageSourceItem implements types.ImageReference.  It's a
+// long-lived object that can only be closed via it's parent MultiImageSource.
+type MultiImageSourceItem struct {
+	*archiveReference
+	tarSource *tarfile.Source
+}
+
+// Manifest returns the tarfile.ManifestItem.
+func (m *MultiImageSourceItem) Manifest() (*tarfile.ManifestItem, error) {
+	items, err := m.tarSource.LoadTarManifest()
+	if err != nil {
+		return nil, err
+	}
+	return &items[0], nil
```

Comment:
So this only allows reading the first image, instead of all/any of them?

Reply:
It points to _one_ image in the tarfile but the tarfile API returns a slice.

Reply:
Oh, this is not `ImageSource.GetManifest`. My mistake.